server/chat-plugins/helptickets.js

```diff
@@ -758,6 +848,165 @@ const pages = {
+			if (FS(`logs/tickets/${nextString}.csv`).readIfExistsSync()) {
+				buttonBar += `<a class="button" href="/view-help-stats-${table}-${nextString}" target="replace" style="float: right">Next Month &gt;</a>`;
+			} else {
+				buttonBar += `<a class="button disabled" style="float: right">Next Month &gt;</a>`;
+			}
+			let col = query.shift() || '';
+
+			let buf = `<div class="pad ladder"><div style="text-align: center">${buttonBar}</div><br />`;
+			buf += `<table style="margin-left: auto; margin-right: auto"><tbody><tr><th colspan="${table === 'tickets' ? 7 : 3}"><h2 style="margin: 5px auto">Help Ticket Stats - ${date.toLocaleString('en-us', {month: 'long', year: 'numeric'})}</h1></th></tr>`;
+			if (table === 'tickets') {
+				if (!['type', 'totaltickets', 'total', 'initwait', 'wait', 'resolution', 'result'].includes(col)) col = 'type';
+				buf += `<tr><th><Button>type</Button></th><th><Button>totaltickets</Button></th><th><Button>total</Button></th><th><Button>initwait</Button></th><th><Button>wait</Button></th><th><Button>resolution</Button></th><th><Button>result</Button></th></tr>`;
+			} else {
+				if (!['staff', 'num', 'time'].includes(col)) col = 'num';
+				buf += `<tr><th><Button>staff</Button></th><th><Button>num</Button></th><th><Button>time</Button></th></tr>`;
+			}
+
+			/** @type {{[key: string]: (string | number)}[]} */
+			let ticketStats = rawTicketStats.split('\n').map(line => {
+				let splitLine = line.split(',');
+				return {
+					type: splitLine[0],
+					total: parseInt(splitLine[1]),
+					initwait: parseInt(splitLine[2]),
+					wait: parseInt(splitLine[3]),
+					resolution: splitLine[4],
+					result: splitLine[5],
+					staff: splitLine[6],
+				};
+			});
+			if (table === 'tickets') {
+				// {total: num, initwait: num, wait: num, resolution: {dead: num, unresolved: num, resolved: num}, result: num positive, totaltickets: num}
+				/** @type {any} */
+				let typeStats = {};
+				for (let stats of ticketStats) {
+					if (!typeStats[stats.type]) typeStats[stats.type] = {total: 0, initwait: 0, wait: 0, resolution: {}, result: 0, totaltickets: 0};
+					let type = typeStats[stats.type];
+					type.totaltickets++;
+					type.total += stats.total;
+					type.initwait += stats.initwait;
+					type.wait += stats.wait;
+					if (['approved', 'valid', 'assisted'].includes(stats.result.toString())) type.result++;
+					if (['dead', 'unresolved', 'resolved'].includes(stats.resolution.toString())) {
+						if (!type.resolution[stats.resolution.toString()]) type.resolution[stats.resolution.toString()] = 0;
+						type.resolution[stats.resolution.toString()]++;
+					}
+				}
+
+				// Calculate averages
+				for (let t in typeStats) {
+					let type = typeStats[t];
+					type.total = Math.round(type.total / type.totaltickets);
```

Comment:
Why are you rounding at each step instead of only at the end?

Reply:
I don't really have a problem with the way it is atm. I wanted to round before sorting to prevent any possible issues there, and the only other place I can think of for rounding would be when building the table itself, which really isnt much different than this, its just all in one line.

Reply:
My concern was that rounding as you calculate the average changes the semantics:

0.49 + 4.49 + 6.49 = 8.98/2 = 3.82, round(3.82) = 4
round(0.49) + round(4.49) + round(6.49) = 10/3 = 3

But I was led astray by the loop and the names (`type.total` really appears like a _total_, not an _average_) - youre doing the sums above (and not rounding) and only computing the averages here (where its correct to round).